src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -75,63 +76,69 @@ internal UnitTestDiscoverer()
                 testElements.Count,
                 source);
 
-            this.SendTestCases(source, testElements, discoverySink);
+            this.SendTestCases(source, testElements, discoverySink, runSettings);
         }
 
-        internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElements, ITestCaseDiscoverySink discoverySink)
+        internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElements, ITestCaseDiscoverySink discoverySink, IRunSettings runSettings)
         {
+            var isDesignMode = RunSettingsUtilities.IsDesignMode(runSettings?.SettingsXml);
```

Comment:
>var isDesignMode = RunSettingsUtilities.IsDesignMode(runSettings?.SettingsXml); [](start = 12, length = 79)

This would be calculated for each source. Should this be part of MSTestSettings instead?